go-controller/pkg/ovn/gateway_init.go

```diff
@@ -339,6 +339,8 @@ func gatewayInit(nodeName string, clusterIPSubnet []*net.IPNet, hostSubnets []*n
 	return nil
 }
 
+// This DistributedGWPort guarantees to always have both IPv4 and IPv6 regardless of dual-stack
```

Comment:
this will break the upgrade though

Reply:
I expect people with more knowledge of ovn than me to chime in, right now it recreates the port to deal with the upgrade and is always dual-stack as the `initJoinLogicalSwitchIPManager`

Reply:
added logic to handle the upgrade